snapshotter/app/service.go

```diff
@@ -56,7 +54,21 @@ func Run(config config.Config) error {
 	group, ctx := errgroup.WithContext(ctx)
 
 	cache := cache.NewSnapshotterCache()
-	snapshotter, err := initSnapshotter(ctx, config, cache)
+	// Start service discovery and metrics proxy monitoring if enabled.
+	var serviceDiscovery *discovery.ServiceDiscovery
+	var monitor *metrics.Monitor
+	if config.Snapshotter.Metrics.Enable {
```

Comment:
- You could declare these variables inside `if` below.
- Do we need to keep Enable=flase case? My worry is that we wouldn't test the path much and could break silently in future.

Reply:
if I declare inside `if` below wouldn't they be out of scope in paths below? e.g. `snapshotter, err := initSnapshotter(ctx, config, cache, monitor)`

Are you suggesting enabling by default/always enabling? I suppose we don't really *need* `Enabled` config, I just am unsure if consumers of this would always want metrics enabled.

Reply:
Oh sorry. Maybe just moving serviceDiscovery inside?

Yes. Always enabling. There might be customers who'd like to disable, but we don't know that yet.

Reply:
Hm ok along with these changes, I can update config to remove disable / enable option for metrics, easy to add back in if that is every requested / required.

In doing so, should address both points

Reply:
Revisiting this, there is a possibility that the in-VM remote snapshotter itself is not configured to emit metrics. Given that's the case, would it make more sense to keep `enable` configuration option so that we aren't attempting to pull metrics that aren't being emitted? @kzys @austinvazquez